persistence/pg.go

```diff
@@ -21,8 +21,9 @@ type Invoice struct {
 
 	SequenceNum       uint64
 	SettleRequestedAt time.Time
-	SettledAt         time.Time
-	Settled           bool
+	ResolutionAt      time.Time
+	Status            types.Status
+	ReachFinalStatus  bool
```

Comment:
Maybe this can be a method `IsFinal()` on `types.Status`? It looks redundant to store in the database.